samples/bluetooth/rssi_nrf53/src/main.c

```diff
@@ -0,0 +1,130 @@
+/* main.c - Application main entry point */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <sys/printk.h>
+#include <sys/util.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_vs.h>
+
+struct rssi_wq_data {
+	struct k_work work;
+	bt_addr_le_t addr;
+	int8_t rssi;
+	uint8_t adv_type;
+} rssi_wq_data;
+
+static int8_t get_chip_temp(void)
+{
+	int err;
+	struct net_buf *rsp;
+	int8_t temperature;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_READ_CHIP_TEMP, NULL, &rsp);
+	if (err) {
+		printk("HCI Error getting chip temp %d\n", err);
+		return -127;
+	}
+
+	struct bt_hci_rp_vs_read_chip_temp *cmd_data = (void *)rsp->data;
+
+	temperature = cmd_data->temps;
+	net_buf_unref(rsp);
+
+	return temperature;
+}
+
+static double pow(double x, double y)
+{
+	double result = 1;
+
+	if (y < 0) {
+		y = -y;
+		while (y--) {
+			result /= x;
+		}
+	} else {
+		while (y--) {
+			result *= x;
+		}
+	}
+
+	return result;
+}
+
+#define ROUND_FL_TO_UINT8(x) (uint8_t)((float)x + 0.5f)
+
+static void compensate_rssi(struct k_work *item)
+{
+	struct rssi_wq_data *wq_data =
+		CONTAINER_OF(item, struct rssi_wq_data, work);
+
+	/* Temperature acquisition time varies between ~700 and ~1500us */
+	int temp = get_chip_temp();
+	uint8_t comp_rssi = (uint8_t)((-1) * wq_data->rssi);
+
+	/* Compute time is ~20us */
+	comp_rssi = ROUND_FL_TO_UINT8(
+		+ (4.9e-5f * pow(comp_rssi, 3))
+		- (9.9e-3f * pow(comp_rssi, 2))
+		+ (1.56f * comp_rssi)
+		- (0.05f * temp)
+		- 7.2f);
+
+	wq_data->rssi = (-1) * (int8_t)comp_rssi;
+
+	if (wq_data->rssi > -70) {
+		char addrstr[BT_ADDR_LE_STR_LEN] = { 0 };
+
+		bt_addr_le_to_str(&wq_data->addr, addrstr, BT_ADDR_LE_STR_LEN);
+
+		printk("Device: %s, RSSI: %d dBm, Temperature: %d C\n",
+			addrstr,
+			wq_data->rssi,
+			temp);
+	}
+}
+
+static void scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+		    struct net_buf_simple *buf)
+{
+	memcpy(&rssi_wq_data.addr, addr, sizeof(bt_addr_le_t));
+	rssi_wq_data.rssi = rssi;
+	rssi_wq_data.adv_type = adv_type;
+	k_work_submit(&rssi_wq_data.work);
+}
+
+void main(void)
+{
+	int err;
+
+	printk("Starting Scanner nRF53 RSSI fix sample\n");
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+	printk("Bluetooth initialized\n");
+
+	/* Initialize work item for compensating RSSI values */
+	k_work_init(&rssi_wq_data.work, compensate_rssi);
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, scan_cb);
+	if (err) {
+		printk("Starting scanning failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Waiting for callbacks\n");
+	while (1) {
+		k_msleep(100);
+	}
```

Comment:
```suggestion

	printk("Scanning successfully started\n");
```
There is no need to keep a while loop at the end of this main function